### PR TITLE
Implement MSC4027 (Custom images in reactions)

### PR DESCRIPTION
**Pull request by surakin**
Fixes https://github.com/element-hq/element-x-android/issues/2159
 
## Type of change

- [x] Feature
- [ ] Bugfix
- [ ] Technical
- [ ] Other :

## Content

Implement rendering of custom images in reaction per MSC4027

## Motivation and context

Ran into mxc reactions while testing Element X and wanted to make it work

## Screenshots / GIFs

image

## Tests

- Send custom reaction using Cinny
- Observe the reaction showing up in Element X

## Tested devices

- [x] Physical
- [x] Emulator
- OS version(s):
Android 6, 13, 14

## Checklist

- [x] Changes have been tested on an Android device or Android emulator with API 23
- [x] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [x] Pull request includes a new file under ./changelog.d. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#changelog
- [x] Pull request includes screenshots or videos if containing UI changes
- [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR

### CODE REVIEW

File: features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/components/MessagesReactionButton.kt [hunk 1]
```diff
@@ -150,13 +156,24 @@ private fun ReactionContent(
...
 ) {
-    Text(
-        text = reaction.displayKey,
-        style = ElementTheme.typography.fontBodyMdRegular.copy(
-            fontSize = 15.sp,
-            lineHeight = reactionEmojiLineHeight,
-        ),
-    )
+    if (reaction.key.startsWith("mxc://")) {
+        BlurHashAsyncImage(
```
Comment (jmartinesp) [inline features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/components/MessagesReactionButton.kt:160]: Same about this one: if it doesn't have a blurhash, `AsyncImage` should be a better option

File: features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/components/reactionsummary/ReactionSummaryView.kt [hunk 2]
```diff
@@ -192,13 +199,24 @@ private fun AggregatedReactionButton(
...
-                text = reaction.displayKey,
-                style = ElementTheme.typography.fontBodyMdRegular.copy(
-                    fontSize = 20.sp,
-                    lineHeight = 25.sp
-                ),
-            )
+            if (reaction.key.startsWith("mxc://")) {
+                BlurHashAsyncImage(
+                    modifier = modifier
+                        .heightIn(min = reactionEmojiLineHeight.toDp(), max = reactionEmojiLineHeight.toDp())
```
Comment (jmartinesp) [inline features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/components/reactionsummary/ReactionSummaryView.kt]: The line height used here should be `25.sp` instead (see the `Text` below). Maybe those should be extracted to a shared private value?

**Review by jmartinesp — CHANGES_REQUESTED** [review]
It's working quite well, thanks! There are a couple of issues that should be fixed before it's merged, though.

File: features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/components/reactionsummary/ReactionSummaryView.kt (hunk 2 above)
Comment (jmartinesp) [reply features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/components/reactionsummary/ReactionSummaryView.kt]: Sorry, I actually meant that it's `25.sp` here while it's `20.sp` in the other file, so each one should have its own value.

File: tests/uitests/src/test/snapshots/images/ui_S_t[f.messages.impl.timeline.components_MessagesReactionButton_null_MessagesReactionButton-Day-11_11_null_0,NEXUS_5,1.0,en].png
Comment (bmarty) [inline tests/uitests/src/test/snapshots/images/ui_S_t[f.messages.impl.timeline.components_MessagesReactionButton_null_MessagesReactionButton-Day-11_11_null_0,NEXUS_5,1.0,en].png:1]: The existing rendering should not change.

**Review by bmarty — CHANGES_REQUESTED** [review]

File: tests/uitests/src/test/snapshots/images/ui_S_t[f.messages.impl.timeline.components_MessagesReactionButton_null_MessagesReactionButton-Day-11_11_null_0,NEXUS_5,1.0,en].png
Comment (jmartinesp) [reply tests/uitests/src/test/snapshots/images/ui_S_t[f.messages.impl.timeline.components_MessagesReactionButton_null_MessagesReactionButton-Day-11_11_null_0,NEXUS_5,1.0,en].png:1]: I believe it's fixed by https://github.com/element-hq/element-x-android/pull/2160/commits/4a6135eed6c0b62a7b8f4083fe3dbd1fc3b7c271 and https://github.com/element-hq/element-x-android/pull/2160/commits/3e3bc7188369de3ad7c582dbc6c4f2b74707b85e

Comment (surakin) [reply tests/uitests/src/test/snapshots/images/ui_S_t[f.messages.impl.timeline.components_MessagesReactionButton_null_MessagesReactionButton-Day-11_11_null_0,NEXUS_5,1.0,en].png:1]: ah yes, screenshots were just being regenerated

**Review by jmartinesp — APPROVED** [review]
Thanks for the changes, LGTM! You might need to add an empty commit at the end so the CI runs again (stupid GH limitations for PRs coming from forks & CI-generated commits, sigh).

**Review by bmarty — APPROVED** [review]
Thanks for the PR @surakin !
Can I ask you to add a comment on this PR with screenshot with the 2 rendering using the latest code:
- timeline message with mxc reaction
- bottom sheet of the details of this image.
Thanks!

**Pull request merged**